monailabel/tasks/infer/vertebra_pipeline.py

```diff
@@ -65,13 +70,28 @@ def __call__(self, request):
         #################################################
         _, result_json_second_stage = self.model_localization_vertebra(second_stage_request)
 
-        # Request for third stage
-        third_stage_request = copy.deepcopy(second_stage_request)
-        third_stage_request["centroids"] = result_json_second_stage
-
         #################################################
         # Run third stage
         #################################################
-        result_file_third_stage, result_json_third_stage = self.model_segmentation_vertebra(third_stage_request)
+        # Request for third stage
+        third_stage_request = copy.deepcopy(second_stage_request)
+        all_outs = {}
+        all_keys = []
+        for centroid in result_json_second_stage["centroids"]:
+            third_stage_request["centroids"] = [centroid]
+            # TO DO:
+            # 1/ Remove the AsDiscrete transform in third stage infer so we get pre-activation outputs
```

Comment:
you can do that based on the request attributes (say skip_xyz) and in the request copy in pipeline you can enable it to skip

Reply:
Do you mean skipping the load image every time?